src/fetch-resources-for-products.js

```diff
@@ -2,6 +2,14 @@ export default function fetchResourcesForProducts(productOrProduct, client) {
   const products = [].concat(productOrProduct);
 
   return Promise.all(products.reduce((promiseAcc, product) => {
+
+    // If the graphql query doesn't find a match, returns null.
+    if (product === null) {
+      promiseAcc.push(null);
```

Comment:
Hi @ilievskizoran - since this `reduce` results in an array of promises, pushing `null` in the array is unnecessary. If the current `product` is null, you can just skip pushing the `fetchAllPages` promises into the array.

Reply:
Hey @melissaluu. You are correct have amended as per the comment.